src/main/java/eu/darkbot/api/managers/HeroItemsAPI.java

```diff
@@ -1,136 +1,46 @@
 /**
  * API to manage in-game items, from ammo, to rockets, abilities or even fireworks.
  *
  * @see SelectableItem
- *
- * TODO: Rethink and refactor this API fully. There are too many methods and not enough flexibility.
  */
 public interface HeroItemsAPI extends API.Singleton {
-
-    /**
-     * This method checks if given {@link Item} can be selected in-game.
-     * If the item isn't in any of the action bars it may not be selectable.
-     *
-     * @param item to check
-     * @return true if item can beb selected
-     */
-    boolean isSelectable(@NotNull Item item);
-
-    /**
-     * @param item to be selected
-     * @return true on successful select
-     */
-    boolean selectItem(@NotNull Item item);
-
-    /**
-     * @param category {@link Category} to be checked
-     * @return true if contains given Category
-     */
-    boolean hasCategory(@NotNull HeroItemsAPI.Category category);
-
-    /**
-     * @return {@link Collection} of all {@link Item}s
-     */
-    Collection<? extends Item> getItems();
-
-    /**
-     * @param category to get item list from
-     * @return list of items associated with given {@link Category}
-     */
-    Collection<? extends Item> getItems(@NotNull HeroItemsAPI.Category category);
-
     /**
-     * Search every {@link Category} for given {@code itemId}.
-     *
-     * @param itemId to be searched for
-     * @return first encounter of given item id
+     * @param selectableItem to get {@link Item} instance
+     * @return {@link Optional<Item>} associated with given {@link SelectableItem}
      */
-    default Optional<Item> findItem(String itemId) {
-        return filterItem(id -> id.equals(itemId));
-    }
+    Optional<Item> getItem(@NotNull SelectableItem selectableItem);
 
     /**
-     * Search {@link Category} for given {@code itemId}.
-     *
-     * @param category to be searched
-     * @param itemId   to be looked for in given category
-     * @return first encounter of given item id
+     * Returns {@link Collection<Item>} of given {@link ItemCategory}
      */
-    default Optional<Item> findItem(@NotNull HeroItemsAPI.Category category, String itemId) {
-        return filterItem(category, id -> id.equals(itemId));
-    }
-
-    default Optional<Item> findItem(@NotNull SelectableItem selectableItem) {
-        Category category = selectableItem.getCategory();
-        return category == null ? findItem(selectableItem.getId()) :
-                findItem(category, selectableItem.getId());
-    }
+    Collection<? extends Item> getItems(@NotNull ItemCategory itemCategory);
 
     /**
-     * Filters items in every {@link Category}
+     * This method checks if given {@link Item} can be used in-game.
+     * If the item isn't in any of the action bars it may not be selectable.
      *
-     * @param filter the items with
-     * @return filtered {@link Optional<Item>} or {@link Optional#empty()}
+     * @param item to check
+     * @return non empty optional if item can be used
```

Comment:
I'm pretty sure there's more to it than just this. if the item doesn't exist it won't be non-empty